### DISCUSSION TITLE
Git 是什麼

### DISCUSSION THREAD

**Discussion by danwan1027** [1101-GS4719-C/Python-Community-GS4719-C#691] [General]
如果你問大部份正在使用 Git 這個工具的人「什麼是 Git」，他們大多可能會回答你「Git 是一種版本控制系統」，專業一點的可能會回答你說「Git 是一種分散式版本的版本控制系統」。「版本控制系統」的英文是「Version Control System」，但這個回答，對沒接觸過的新手來說，有講跟沒講差不多。
簡單來說，工程師每天都要對code做很多的修改，甚至想要回到修改前的樣子重新進行編輯，如果沒有一個工具來紀錄每次更改的內容，想要回到之前的版本就會變得相當困難，因此，git 的出現解決了這個煩惱，每次修改完程式並 push 上 git，他就能完美紀錄每次更改的細節，也能附上每次修改的備註。

**Comment by a45320614** [comment]
想知道Git是甚麼嗎? 快來參加GDSC 9/27的社課吧!

**Reply by danwan1027** [reply]
太狠了 偷偷推銷

**Comment by Yunhans** [comment]
這篇git文章是@danwan1027 送給大家的gift

**Reply by evanchou1204** [reply]
酷酷的諧音梗😎

**Comment by Chiway1229** [comment]
哇 謝謝你的分享
上周我還在納悶git是做什麼的

**Comment by StevenWanglolz** [comment]
git聽起來真的好方便喔! 像我之前在打code時發現我的方向完全錯誤，所以需要砍掉大部分的code，如果可以利用git紀錄之前的進度就不會這麼麻煩了~

**Comment by delia07152** [comment]
還有一種版控是CVCs，因為專案通常會由許多人完成，而CVCs除了有備份還多了共用的功能。不過兩著共同的缺點是，他們都是在自己電腦上備份，所以一旦本地的設備發生狀況，基本上就回不去了。而Git改善了這個缺點，它將資料分散在不同設備，而這類型的版控稱為DVCs。

**Comment by dabbie109303501** [comment]
補充 : 除了課堂上教的用vscode push project，也可以直接開cmd(命令提示字元)，移動路徑到檔案位置
1. 輸入git add .
2. 輸入git commit -m "更新資訊"
3. 輸入git push ssh@key
就可以更新版本囉~

*    ssh@key可以在要push的專案上的這裡找到
![119701](https://user-images.githubusercontent.com/75252877/192017334-6176803d-d3b4-45f6-a93f-24346369341d.jpg)

**Comment by evanchou1204** [comment]
確實是很人性化的一個工具，十分有用

**Comment by Pietra1226** [comment]
好難懂喔...不知道是不是我理解能力有障礙，每次理解一個新的語言或是架構都要花很久時間

**Comment by Tuoresa** [comment]
原來git所代表的是如此實用的功能，真的是程式編寫的一大福音！感謝分享！

**Comment by ru0372572** [comment]
Git的快照(Snap)功能一開始真的聽不太懂，查了資料和聽了版主的解釋，能了解為什麼有這個功能了！

**Comment by poiu6fi** [comment]
之前做專題的時候因為對git不熟悉把github上面的東西弄炸了，很對不起組員們，以後要找時間好好學一下git了

**Comment by OHoln** [comment]
要成為一個合格的軟體工程師，git真的是一門必修課，要是現在不抓緊機會好好學，未來面對超大型的程式時想必會是一個大災難。

**Comment by 710388** [comment]
非常好用，不論是在下載模組還是更新套件的時候

**Comment by tshsu04224** [comment]
最近因為網頁程式設計課，跟小組一起做一個網頁，真的十分頻繁的在使用git進行共編，我覺得git真的是一個很便利的工具，有版本紀錄，寫壞了可以回溯，也可以大幅的增加小組合作的效率，真的是寫程式時的一大福音！

**Comment by 155Julia** [comment]
我這學期很多網頁程式設計課，和小組一起做一個網頁，因此也頻繁地使用 Git 進行共編。我覺得 Git 真的是一個很便利的工具，有版本紀錄，寫壞了可以回溯，也能大幅增加小組合作的效率，真的是寫程式時的一大福音！但每次合併時常常會有衝突，這時候就覺得很麻煩，又要重新查指令才能整合程式碼。不過，即使有這些麻煩，我仍然覺得 Git 是一個不可或缺的工具，能夠大大提升我們的開發效率和協作體驗。
